src/main/scala/de/bripkens/ha/AppActor.scala

```diff
@@ -1,16 +1,26 @@
 package de.bripkens.ha
 
-import akka.actor.{Props, ActorLogging, Actor}
+import akka.actor.{Actor, ActorLogging, Props}
 import com.fasterxml.jackson.databind.ObjectMapper
+import de.bripkens.ha.reporting.{ConsoleReporter, SlackReporter}
+
+object AppActor {
+
+  final val Name = "app"
+
+  def props(mapper: ObjectMapper, config: Configuration) = Props(new AppActor(mapper, config))
+}
 
 class AppActor(val mapper: ObjectMapper, val config: Configuration) extends Actor
                                                                     with ActorLogging {
 
-  val reporters = config.reporters.map { case (name, config) =>
-    val actor = context.actorOf(
-      Props(config.implementation, mapper, config),
-      s"reporter-$name"
-    )
+  val reporters = config.reporters.map{ case (name, reporterConfig) => 
+    val props = reporterConfig match {
+      case config@ConsoleReporterConfig(_)           => ConsoleReporter.props(mapper, config)
```

Comment:
I suggest to use the typed pattern here, i.e. `config: ConsoleReporterConfig`, because you don't destruct the case class.